BME-79: Observation / Survey enhancements (brac)

sagarkoshti1990 opened the pull request:
https://mantra4change-team-yliooeqr.atlassian.net/jira/software/projects/BME/boards/34?selectedIssue=BME-79

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

## Release Notes

* **New Features**
  * File upload capability for questionnaire attachments with file preview, validation, and size limit enforcement
  * Grid-based card layout option for improved question display and organization
  * Section tabs for managing multi-section questionnaires with organized navigation
  * Enhanced pagination configuration with responsive layout options

* **Chores**
  * Updated package metadata and versioning

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

syedameenabrar committed:
continue btn added in domains page

syedameenabrar committed:
spinner and other css issues which face in pwa

syedameenabrar committed:
removed token

syedameenabrar committed:
observe again

syedameenabrar committed:
changed static to dynamic value

syedameenabrar committed:
removed consoles

syedameenabrar committed:
renamed the variable

syedameenabrar committed:
deleted unwanted component and services

syedameenabrar committed:
setconfig removed

5Amogh committed:
Merge pull request #30 from syedameenabrar/samiksha-section-listing

Observation listing and report

5Amogh committed:
version change and npm publish

syedameenabrar committed:
no data img styling, add and cancle btn enable

syedameenabrar committed:
version updated

syedameenabrar committed:
Merge branch 'observation-1.0.0' of https://github.com/5Amogh/questionnaire-player into samiksha-section-listing

syedameenabrar committed:
version updated

5Amogh committed:
Merge pull request #31 from syedameenabrar/samiksha-section-listing

observation listing

syedameenabrar committed:
multiple observations

syedameenabrar committed:
multiple observation filter in reports

syedameenabrar committed:
version uprade

syedameenabrar committed:
loader for pdf

syedameenabrar committed:
report page changes

5Amogh committed:
Merge pull request #32 from syedameenabrar/samiksha-section-listing

Multiple Observation Filter In Reports

syedameenabrar committed:
issues fixes

syedameenabrar committed:
version upgrade

5Amogh committed:
Merge pull request #34 from syedameenabrar/samiksha-section-listing

issues fixes

syedameenabrar committed:
search input clear button and fixed bottom load more button

syedameenabrar committed:
removed backService and navigation from list to reports code added

syedameenabrar committed:
report listing in listing component

syedameenabrar committed:
charts.js issues fixed

syedameenabrar committed:
report filter with model issue fixed

syedameenabrar committed:
load more button in listing

syedameenabrar committed:
loadmore issue

syedameenabrar committed:
removed token, comments

syedameenabrar committed:
reports filter UI

syedameenabrar committed:
label rename changes

5Amogh committed:
Merge pull request #35 from syedameenabrar/samiksha-section-listing

Report Charts issue fixed

syedameenabrar committed:
removed date pipe

5Amogh committed:
Merge pull request #36 from syedameenabrar/samiksha-section-listing

removed date pipe

syedameenabrar committed:
css issues

syedameenabrar committed:
eveidence preview

syedameenabrar committed:
evidence

syedameenabrar committed:
uninstalled jwt-decoder

syedameenabrar committed:
package update

5Amogh committed:
Merge pull request #37 from syedameenabrar/samiksha-section-listing

Observation issue release-1

syedameenabrar committed:
offline and online feature enabled

syedameenabrar committed:
package update

5Amogh committed:
Merge pull request #38 from syedameenabrar/samiksha-section-listing

offline and online feature enabled

syedameenabrar committed:
domain report started

syedameenabrar committed:
domain wise chart done

syedameenabrar committed:
token regeneration and domain wise report

syedameenabrar committed:
multiple select reports

syedameenabrar committed:
domain view and criteria styling in reports

syedameenabrar committed:
report dates and status

syedameenabrar committed:
package updated

syedameenabrar committed:
removed unwanted code in reports

5Amogh committed:
Merge pull request #39 from syedameenabrar/samiksha-section-listing

domain reports and creatria UI issues

syedameenabrar committed:
reports offline data, footer btn fixed ui, add entity pass parentId

syedameenabrar committed:
observe again and reports view butn hide and show

syedameenabrar committed:
isRubricDriven reports for multiple submisson

syedameenabrar committed:
edit and delete options for multiple domains

syedameenabrar committed:
date formating in reports fixed

syedameenabrar committed:
package updated

5Amogh committed:
Merge pull request #40 from syedameenabrar/samiksha-section-listing

Form edit disable, delete & edit btn for multiple rubric, Offline msg in reports, observe again & report view btn

syedameenabrar committed:
domain wise report in report lisitng page

syedameenabrar committed:
load more btn UI fixed

syedameenabrar committed:
filter pdf in reports

5Amogh committed:
Merge pull request #41 from syedameenabrar/samiksha-section-listing

Filter data in reports pdf

syedameenabrar committed:
date pipe issue resolved

syedameenabrar committed:
package update

5Amogh committed:
Merge pull request #42 from syedameenabrar/samiksha-section-listing

date pipe issue resolved

syedameenabrar committed:
get search entities and reset data in search entities once deleted

syedameenabrar committed:
marke checked if data already presend in serarch entity

syedameenabrar committed:
version update

5Amogh committed:
Merge pull request #43 from syedameenabrar/samiksha-section-listing

Search entities and add entities

syedameenabrar committed:
hide and show save and submit btns in questioner player

syedameenabrar committed:
version update

5Amogh committed:
Merge pull request #44 from syedameenabrar/samiksha-section-listing

hide and show save and submit btns in questioner player

syedameenabrar committed:
add entities check box issue fixed

5Amogh committed:
Merge pull request #45 from syedameenabrar/samiksha-section-listing

add entities check box issue fixed

syedameenabrar committed:
horizontal sroll for reports data

syedameenabrar committed:
sticky footer in section page

syedameenabrar committed:
back to section lisiting

syedameenabrar committed:
style for sesion list, enabled scroller bar in reports

5Amogh committed:
Merge pull request #46 from syedameenabrar/samiksha-section-listing

Back to section page

syedameenabrar committed:
entity type from listing page and disbale add button if no error comes

syedameenabrar committed:
save pop up functionality

syedameenabrar committed:
boolean value changes

syedameenabrar committed:
updated version

syedameenabrar committed:
spaces

syedameenabrar committed:
spaces

syedameenabrar committed:
removed two if condition

syedameenabrar committed:
profile ui issue, select single entity

5Amogh committed:
Merge pull request #47 from syedameenabrar/samiksha-section-listing

Pop up save functionality

syedameenabrar committed:
firefox ui issue, sections page popup back button issue

syedameenabrar committed:
removed console

5Amogh committed:
Merge pull request #48 from syedameenabrar/samiksha-section-listing

Firefox load button ui issue, section back navigation pop issue

syedameenabrar committed:
limit 50 mb file upload, height adjust in listing page , ngDestory and confirm it is observation variable

syedameenabrar committed:
auto save when navigate back

syedameenabrar committed:
matrix question render

syedameenabrar committed:
matrix ui 1

syedameenabrar committed:
pacakage updated

syedameenabrar committed:
matrix ui com

syedameenabrar committed:
upgrade version to 19

5Amogh committed:
Merge pull request #49 from syedameenabrar/samiksha-section-listing

Save pop issue and ui allignment reports

syedameenabrar committed:
removed formObservation input decorator and servie data

5Amogh committed:
Merge pull request #50 from syedameenabrar/samiksha-section-listing

removed formObservation input decorator and servie data

5Amogh committed:
section listing routing issue fix

syedameenabrar committed:
url decodeURIComponent issue fixed

syedameenabrar committed:
version update

syedameenabrar committed:
upload evidence for asure cloud added extra header

syedameenabrar committed:
router back issue  solved

syedameenabrar committed:
package version updated

5Amogh committed:
Merge pull request #51 from syedameenabrar/observation-1.1.0

url decodeURIComponent issue fixed

5Amogh committed:
style bug resolved

5Amogh committed:
Merge branch 'observation-1.1.0' of https://github.com/5Amogh/questionnaire-player into observation-1.1.0

5Amogh committed:
version update

5Amogh committed:
survey q-map re-dir and header fix

5Amogh committed:
observation with rubrics UI aligment fix

syedameenabrar committed:
important tag added at line number 179 as Ram was facing issue in his portal

syedameenabrar committed:
package version updated

syedameenabrar committed:
Merge remote-tracking branch 'upstream/observation-1.1.0' into observation-1.1.0

syedameenabrar committed:
one css code line changes in reports css page a 155

syedameenabrar committed:
removed "/" at beggining for no data image path, as image is not showing

syedameenabrar committed:
npm package version updated to v-4.1.10

syedameenabrar committed:
download file package installation

syedameenabrar committed:
downloadFile code changed

syedameenabrar committed:
css code commented in listng css page as its creating ui issue in Ram project

syedameenabrar committed:
replased "/assets" tio "assets" in listing page

syedameenabrar committed:
domain heading & save popup buttons name change

Naveen8238 committed:
Questionnaire player chanages to save and indexDB changes

syedameenabrar committed:
Merge pull request #1 from codeNaveen1234/observation-1.1.0

Questionnaire player chanages to save and indexDB changes

syedameenabrar committed:
Merge branch 'observation-1.1.0' of https://github.com/syedameenabrar/questionnaire-player into observation-1.1.0

syedameenabrar committed:
package version updated to 4.1.18

Naveen8238 committed:
Questionnaire player changes

Naveen8238 committed:
Package versions changes

syedameenabrar committed:
survey expired link

syedameenabrar committed:
Merge pull request #2 from codeNaveen1234/observation-1.1.0

Questionnaire player changes

syedameenabrar committed:
package update to v4.1.20

syedameenabrar committed:
diksha changes implemented and evidence issue fixed

syedameenabrar committed:
attachment hint model and back navigation once form is submitted

syedameenabrar committed:
removed save button and save data on every change  and also fixed Ui issues

syedameenabrar committed:
auto save, removed save button

syedameenabrar committed:
auto save issue in survery fixed

syedameenabrar committed:
matrix question issue resolved

syedameenabrar committed:
cards css issue fixed

syedameenabrar committed:
matrix save form issue fixed

syedameenabrar committed:
question-map navigation issue fixed

syedameenabrar committed:
privacy policy changed to elevate

syedameenabrar committed:
evidenceCode issue resolved for survery db data as undefined

syedameenabrar committed:
4.1.31

syedameenabrar committed:
page wise status bar

syedameenabrar committed:
4.1.32

syedameenabrar committed:
version updated to 4.1.32

syedameenabrar committed:
page complete logic rewritten

syedameenabrar committed:
4.1.33

syedameenabrar committed:
privacy policy ui updated

syedameenabrar committed:
audio formate updated

syedameenabrar committed:
4.1.34

syedameenabrar committed:
attachment message updated

syedameenabrar committed:
4.1.35

syedameenabrar committed:
save button for normal flow added

syedameenabrar committed:
program survey not saving issue resolved  - 3807

syedameenabrar committed:
4.1.36

syedameenabrar committed:
build generated

Naveen8238 committed:
questionnaire player changes

syedameenabrar committed:
Merge pull request #6 from codeNaveen1234/observation-1.1.0

questionnaire player changes

syedameenabrar committed:
3959 - Incorrect Criteria Name Displayed When Accessing Observation via Deep Link

syedameenabrar committed:
4.1.39

syedameenabrar committed:
4013 - Observation without rubric inapp report is not getting generated.

syedameenabrar committed:
4.1.40

syedameenabrar committed:
4122 - Duplicate/Conflicting Toast Messages After Observation Submission

syedameenabrar committed:
4121 - Entire Observation Page is Scrollable, Hiding 'Start' Button

syedameenabrar committed:
4.1.41

syedameenabrar committed:
4120, 4118 - issue resolved

syedameenabrar committed:
4.1.42

syedameenabrar committed:
4128 - Survey Deep Link Incorrectly Displays 'Start' Button.

syedameenabrar committed:
4132 - Survey tile not showing ‘In Progress’ status after providing some responses via deep link

syedameenabrar committed:
4.1.43

syedameenabrar committed:
4150 - Observation Progress Bar Resets After Navigating Away and Returning

syedameenabrar committed:
4139 - question map UI is different from saas prod and flutter

syedameenabrar committed:
4.1.44

syedameenabrar committed:
4120 - Observation without rubric: Questionnaire Auto-Enables Without Clicking 'Start'

syedameenabrar committed:
4.1.45

syedameenabrar committed:
4118 - Observation with rubric Flow: 'Start' Button Reappears After Observation Has already Begun, || and hide submit button when showing start button

syedameenabrar committed:
4.1.46

syedameenabrar committed:
range input issue resolved

syedameenabrar committed:
4.1.47

syedameenabrar committed:
4180 - Deep Link Access Issue for Observation With Rubric on Second Domain

syedameenabrar committed:
4.1.48

syedameenabrar committed:
4188 - Submit Button Disappears and Questionnaire Resets After Navigating Through Map

syedameenabrar committed:
4.1.49

syedameenabrar committed:
removed unwanted comments

syedameenabrar committed:
4199 - Start Button Incorrectly Enabled for Non-Targeted Users Accessing Observation Deep Links

syedameenabrar committed:
4198 - Deep Link Navigation Shows Blank Page for a Moment Instead of Loader for Domain 2

syedameenabrar committed:
4198 - Deep Link Navigation Shows Blank Page for a Moment Instead of Loader for Domain 2

syedameenabrar committed:
4.1.50

Commit:
fix:tabs

Commit:
fix:input grid group section

Commit:
fix: improve input grid layout and functionality

Commit:
fix:[FE] Observation Form — Validations and Submission Flow (as per BRAC use case)

Commit:
fix:[FE] New Field Type – File Upload + [FE] Label + Input Horizontal Layout

Commit:
Merge branch 'observation-1.1.0' into brac-release-3.0.0

sagarkoshti1990 commented:
@coderabbitai review

Commit:
[FE] Support Passing Data Directly to Web Component and Bypass API Calls

Commit:
[FE] New Field Type – Entity Dropdown (Dynamic API-Based Dropdown with Search & Pagination)

Vinod-V3 commented on projects/questionnaire-app/src/app/mockData.json:
this file can be deleted, no need to commiting it
If this file is required for further development as of now, keep the file for now but please delete the contents
That u can just keep it in local
It has more than 3000 lines

sagarkoshti1990 replied:
okay

Commit:
fix:search

Commit:
fix: add progress post message

Commit:
feat: add select-dropdown and multiselect-dropdown field types

Commit:
fix:add api config showSaveDraftButton

Commit:
ix: validation summary page — on button/link click, focus the field; add UI/UX when showSaveDraftButton is true

Commit:
add web component build and css

Commit:
fix: progress input and page level

Commit:
fix:tabs lable issue

Commit:
fix: upload file preview on submitted form

Commit:
fix: allow form submission when showSaveDraftButton is true and required fields are filled

Commit:
fix:UIUX Alert component

Commit:
issue on submition setApiService

Commit:
fix:submissionSuccess in sendMessage

Commit:
fix: set defaultValues with readOnly field true/false

Commit:
fix(entity-dropdown): use responseType to determine multi-select mode and simplify confirmSelection
- Change isMultiSelect to check question.responseType === 'multiselect' instead of metaInformation config
- Simplify confirmSelection method by removing unnecessary try-catch wrapper
- Ensure form control value is set correctly for single-select radio button selection
- Fix modal closing and value setting issues for entity dropdown component

Commit:
fix:UIUX and hide lib-checkbox-input for entity dropdown

Commit:
fix:baseUrl error on server

Commit:
fix:submit valid issue and lib-entity-dropdown-input issue

Commit:
fix: validation change and add config showPrivacyPopup for upload

Commit:
fix:uiux

Commit:
fix:uiux

Commit:
fix:entity dropdown UIUX

Commit:
Fix:entity dropdown add Button position.

Commit:
Fix: Add display: none by default to the span when !question.visibleIf?.length || question.canDisplay === true.

Commit:
Fix: Implement min, max, minLength, and maxLength validations. For optional fields with additional validations (such as min/max, minLength/maxLength, or regex), display the validation popup for the respective field.

Commit:
fix:dynamicEntityTyperequireDynamicAnswers lableMapping

Commit:
Fix: Improve UI/UX for entity dropdown and remove save progress on ngOnDestroy.

Commit:
Fix: Made showToast configuration-driven and always send via window.postMessage

Commit:
feat: Add data preview in matrix question cards

- Display first 5 filled fields in matrix instance cards
- Show actual field names from question data
- Add 'more' indicator when instance has more than 5 fields
- Clicking card opens full popup with all fields (existing behavior)
- Built web component with latest changes

Commit:
fix:matrix card value options show + calculateInitialProgress issue

Commit:
Fix: Added visibleIf operator “OR” option with value as an array or comma-separated string

Commit:
Revert "Fix: Added visibleIf operator “OR” option with value as an array or comma-separated string"

This reverts commit 107ba4bb8bc72c61e6996a555e88d4dfc53ab188.

Commit:
Fix: Added icon to question group title

Commit:
fix:add config saveProgressStorageType

Commit:
fix: set default value only on first load

Commit:
fix:uiux

Commit:
fix: set default value when local value is empty, undefined, or null

Commit:
fix:footer button css change

Commit:
fix: add No changes to save. for save draft

Commit:
fix: Clicking the Next tab button scrolls top

Commit:
fix: handle postMessage for native WebView app as well

Commit:
fix:subHeader

Commit:
fix: remove extra spaces before questions and handle spacing when question number is missing

Commit:
fix: set textarea default rows to 3 with auto height adjustment

Commit:
fix:upload file donwload issue

Commit:
fix: add offline flag
